imaginairy/schema.py

```diff
@@ -172,385 +133,543 @@ def handle_b64(value: Any) -> "LazyLoadingImage":
+    #     if info.data.get("init_image") is None and v:
+    #         msg = "You must set `init_image` if you want to use a mask"
+    #         raise ValueError(msg)
+    #     return v
 
-        return self
+    # @field_validator("model", mode="before")
+    # def set_default_diffusion_model(cls, v):
+    #     if v is None:
+    #         return config.DEFAULT_MODEL
 
-    @field_validator("height", "width")
-    def validate_image_size(cls, v, info: core_schema.FieldValidationInfo):
-        from imaginairy.model_manager import get_model_default_image_size
+    #     return v
 
-        if v is None:
-            v = get_model_default_image_size(info.data["model"])
+    # @field_validator("fix_faces_fidelity", mode="before")
+    # def validate_fix_faces_fidelity(cls, v):
+    #     if v is None:
+    #         return 0.2
 
-        return v
+    #     return v
 
-    @field_validator("caption_text", mode="before")
-    def validate_caption_text(cls, v):
-        if v is None:
-            v = ""
+    # @field_validator("sampler_type", mode="after")
+    # def validate_sampler_type(cls, v, info: core_schema.FieldValidationInfo):
+    #     from imaginairy.samplers import SamplerName
 
-        return v
+    #     if v is None:
+    #         v = config.DEFAULT_SAMPLER
 
-    @property
-    def prompts(self):
-        return self.prompt
+    #     v = v.lower()
+
+    #     if info.data.get("model") == "SD-2.0-v" and v == SamplerName.PLMS:
+    #         raise ValueError("PLMS sampler is not supported for SD-2.0-v model.")
+
+    #     if info.data.get("model") == "edit" and v in (
+    #         SamplerName.PLMS,
+    #         SamplerName.DDIM,
+    #     ):
+    #         msg = "PLMS and DDIM samplers are not supported for pix2pix edit model."
+    #         raise ValueError(msg)
+    #     return v
+
+    # @field_validator("steps")
+    # def validate_steps(cls, v, info: core_schema.FieldValidationInfo):
+    #     from imaginairy.samplers import SAMPLER_LOOKUP
+
+    #     if v is None:
+    #         SamplerCls = SAMPLER_LOOKUP[info.data["sampler_type"]]
+    #         v = SamplerCls.default_steps
+
+    #     return int(v)
+
+    # @model_validator(mode="after")
+    # def validate_init_image_strength(self):
+    #     if self.init_image_strength is None:
+    #         if self.control_inputs:
+    #             self.init_image_strength = 0.0
+    #         elif self.outpaint or self.mask_image or self.mask_prompt:
+    #             self.init_image_strength = 0.0
+    #         else:
+    #             self.init_image_strength = 0.2
+
+    #     return self
+
+
+@dataclass
+class ImaginePrompt:
+    prompt: list[WeightedPrompt]
+    negative_prompt: list[WeightedPrompt]
+    prompt_strength: float
+    init_image: InitImageConfig | None
+    width: int
+    height: int
+    seed: int
+    steps: int
+    model: str
+    model_config: Path  # ?: what is this?
```

Comment:
in non-refiners world everyone uses those yaml config files. Since there are various architectures (1.5, 2.1, 2.1v, etc) if someone puts in custom weights into the `model` parameter then they may need to specify what kind of weights they are.   Probably `model` and `model_config` would be better labeled as `weights`  and `model_architecture`